presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -475,7 +475,7 @@ private static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inpu
             }
             if (containsEligibleTableBucket) {
                 LocatedFileStatus file = files.get(partitionBucketNumber);
-                splitFactory.createInternalHiveSplit(file, readBucketNumber)
+                splitFactory.createInternalHiveSplit(file, readBucketNumber, partitionBucketNumber)
```

Comment:
Correct me if I am wrong, but I think we should use `tableBucketNumber` for `$bucket`? (See also comment on line  457)

Reply:
That is correct. Fixing it